subsys/bluetooth/host/cs.c

```diff
@@ -360,72 +543,184 @@ void bt_hci_le_cs_subevent_result(struct net_buf *buf)
+	p_result->header.num_antenna_paths = evt->num_antenna_paths;
+	p_result->header.num_steps_reported = evt->num_steps_reported;
+
+	p_result->header.config_id = 0;
+	p_result->header.start_acl_conn_event = 0;
+	if (conn) {
+		p_result->header.config_id = evt->config_id;
+		p_result->header.start_acl_conn_event =
+			sys_le16_to_cpu(evt->start_acl_conn_event_counter);
 	}
 
-#if defined(CONFIG_BT_CHANNEL_SOUNDING_TEST)
-	if (sys_le16_to_cpu(evt->conn_handle) == BT_HCI_LE_CS_TEST_CONN_HANDLE) {
-		result.header.config_id = 0;
-		result.header.start_acl_conn_event = 0;
+	if (evt->subevent_done_status != BT_HCI_LE_CS_SUBEVENT_DONE_STATUS_PARTIAL) {
+		invoke_subevent_result_callback(conn, p_result);
+	}
+
+	if (evt->procedure_done_status != BT_CONN_LE_CS_PROCEDURE_INCOMPLETE) {
+		/* We can now clear the any reassembly buffer allocated for this procedure,
+		 * to avoid code duplication, we're using the abort label to do so
+		 */
+		goto abort;
+	}
+
+	if (conn) {
+		bt_conn_unref(conn);
+		conn = NULL;
+	}
 
-		cs_test_callbacks.le_cs_test_subevent_data_available(&result);
+	return;
+
+abort:
+	if (conn) {
+		bt_conn_unref(conn);
+		conn = NULL;
+	}
+
+	reassembly_buf = get_reassembly_buf(conn_handle, false);
+	if (reassembly_buf) {
+		free_reassembly_buf(&reassembly_buf);
+	}
+}
+
+void bt_hci_le_cs_subevent_result_continue(struct net_buf *buf)
+{
+	struct bt_conn *conn = NULL;
+	struct bt_hci_evt_le_cs_subevent_result_continue *evt;
+	struct net_buf *reassembly_buf = NULL;
+	uint16_t conn_handle;
 
+	if (buf->len < sizeof(*evt)) {
+		LOG_ERR("Unexpected end of buffer");
+		return;
+	}
+
+	evt = net_buf_pull_mem(buf, sizeof(*evt));
+	conn_handle = sys_le16_to_cpu(evt->conn_handle);
+
+#if defined(CONFIG_BT_CHANNEL_SOUNDING_TEST)
+	if (conn_handle == BT_HCI_LE_CS_TEST_CONN_HANDLE) {
+		if (!cs_test_callbacks.le_cs_test_subevent_data_available) {
+			LOG_WRN("No callback registered. Discarded subevent results from CS Test.");
+			return;
+		}
 	} else
 #endif /* CONFIG_BT_CHANNEL_SOUNDING_TEST */
 	{
-		result.header.config_id = evt->config_id;
-		result.header.start_acl_conn_event =
-			sys_le16_to_cpu(evt->start_acl_conn_event_counter);
+		conn = bt_conn_lookup_handle(conn_handle, BT_CONN_TYPE_LE);
+		if (!conn) {
+			LOG_ERR("Unknown connection handle when processing subevent results");
+			return;
+		}
+	}
 
-		notify_cs_subevent_result(conn, &result);
+	uint16_t step_data_len = evt->num_steps_reported ? buf->len : 0;
 
+	reassembly_buf = continue_reassembly(conn_handle, evt->steps, step_data_len);
+	if (!reassembly_buf) {
+		goto abort;
+	}
+
+	reassembled_result.header.procedure_done_status = evt->procedure_done_status;
+	reassembled_result.header.subevent_done_status = evt->subevent_done_status;
+	reassembled_result.header.procedure_abort_reason = evt->procedure_abort_reason;
+	reassembled_result.header.subevent_abort_reason = evt->subevent_abort_reason;
+	reassembled_result.header.num_antenna_paths = evt->num_antenna_paths;
```

Comment:
It's not 100% clear what should the host do with this information as it won't be part of the ranging data. Following the assumption that, we could receive events where this can be zero or non-zero, we'll accept whatever non-zero value passed.

Though looks like I forgot to remove this line. Will do that now.